pkg/trace/writer/sender.go

```diff
@@ -326,6 +334,32 @@ func (s *sender) recordEvent(t eventType, data *eventData) {
 	s.cfg.recorder.recordEvent(t, data)
 }
 
+// isEnabled returns true if the sender is enabled. Non-MRF senders are always enabled, and MRF ones
+// only when isMRFEnabled is true
+func (s *sender) isEnabled() bool {
+	if !s.cfg.isMRF {
+		return true
+	}
+	if s.cfg.isMRFEnabled == nil {
+		log.Errorf("Error checking MRF. isMRFEnabled() is not configured on domain %v", s.cfg.url)
+		return true
+	}
+	if s.cfg.isMRFEnabled() {
+		if !s.enabled {
+			log.Infof("Sender for domain %v has been failed over to, enabling it for MRF.", s.cfg.url)
+			s.enabled = true
```

Comment:
I think we can reduce the level of nesting here by setting s.enabled and then just returning true here?